visualizer/src/service/tools/segment/thresholdMachine.js

```diff
@@ -1,52 +1,52 @@
 import { assign, Machine, sendParent } from 'xstate';
 import { toolActions, toolGuards } from './toolUtils';
 
-const createThresholdMachine = ({ x, y, foreground }) =>
-  Machine(
-    {
-      initial: 'idle',
-      context: {
-        x,
-        y,
-        foreground,
-        firstPoint: [0, 0],
-      },
-      states: {
-        idle: {
-          on: {
-            mousedown: { target: 'dragging', actions: 'saveFirstPoint' },
-          },
-        },
-        dragging: {
-          on: {
-            mouseup: { target: 'idle', actions: 'threshold' },
-          },
+const thresholdMachine = Machine(
+  {
+    initial: 'idle',
+    context: {
+      x: null,
+      y: null,
+      foreground: null,
+      firstPoint: [null, null],
+    },
+    states: {
+      idle: {
+        on: {
+          mousedown: { target: 'dragging', actions: 'saveFirstPoint' },
         },
       },
-      on: {
-        COORDINATES: { actions: 'setCoordinates' },
-        FOREGROUND: { actions: 'setForeground' },
+      dragging: {
+        on: {
+          mouseup: { target: 'idle', actions: 'threshold' },
+        },
       },
     },
-    {
-      guards: toolGuards,
-      actions: {
-        ...toolActions,
-        saveFirstPoint: assign({ firstPoint: ({ x, y }) => [x, y] }),
-        threshold: sendParent(({ foreground, firstPoint, x, y }, event) => ({
-          type: 'EDIT',
-          action: 'threshold',
-          args: {
-            x1: firstPoint[0],
-            y1: firstPoint[1],
-            x2: x,
-            y2: y,
-            // frame: context.frame,
-            label: foreground,
-          },
-        })),
-      },
-    }
-  );
+    on: {
+      EXIT: '.idle',
```

Comment:
Why does the Brush and Threshold machine need the `EXIT` event, but not the Flood, Select, Trim, or Watershed?

Reply:
The brush and threshold machines have internal finite state that we need to reset when changing tools midway through using them. The other machines also receive an EXIT event when switching tools, but they don't need to use them.

For example, if we start painting a brush stroke, then switch tools, if we did not send the EXIT event, then when we start using the brush tool again, we would still be painting that same brush stroke

Reply:
Actually now that I say that, watershed needs to use EXIT because it has two steps